### ISSUE TITLE
[Feature Request]: More Assumption checks like Shapiro-Wilk in anova & regression

### ISSUE THREAD

**Issue by TarandeepKang** [jasp-stats/jasp-issues#2878]
### Description

Enhance existing normality testing and mirror functionality available in other modules

### Is your feature request related to a problem?

Currently normality testing can be done graphically some may prefer an additional numerical test

### Is your feature request related to a JASP module?

Regression

### Describe the solution you would like

Consider implementing Shapiro-Wilk test across regression and consider consistent naming and organisation of assumption checks across regression module and t-test

**Comment by TarandeepKang** [comment]
A follow-up question, if I may? You have recently implemented the improved version in MANOVA. Would it be worth also implementing this version everywhere the current multivariate Shapiro-Wilk is used (i.e. correlation et cetera)?

**Comment by tomtomme** [comment]
Transfering my list with what we have and what we miss in terms of assumption checks from 
https://github.com/jasp-stats/jasp-issues/issues/490#issuecomment-1901141226


Variance tests
- [x] Breusch-Pagan's + White's p-value for testing H0: homoscedastic residuals in Regressions
- [x] Levenes-Test of Median & Trimmed Mean for ANOVAs 
Textbook https://www.itl.nist.gov/div898/handbook/eda/section3/eda35a.htm
In R: https://search.r-project.org/CRAN/refmans/lawstat/html/levene.test.html

Normality Tests
- [X] Kolmogorov-Smirnov within Distribution module
- [x] Shapiro-Wilk in Correlation & Descriptives & Manova & Distribution module
- [x] QQ-Plots within RM ANOVA & tTests & Regressions
- [x] 95% CIs for QQ-Plot in Distributions module
- [x] Kolmogorov-Smirnov & shapiro-wilk within (RM) AN(C)OVA (& Regression, however there we can now export residuals to test in distribution module)
- [ ] QQ-Plot within MANOVA 
- [x] 95% CIs for QQ-Plot in Regressions &  (RM) AN(C)OVAs

Linearity Test (logistic)
- [ ] assumption of "linearity in the logit" via 
Box & Tidwell, 1962 or 
Hosmer & Lemeshow, 1989 or 
Menard, 2002, 2010 or 
le Cessie-van Houwelingen-Copas-Hosmer 

Box-Tidwell
https://stackoverflow.com/questions/66618959/how-to-run-a-box-tidwell-test-in-r-to-test-for-a-linear-relationship-between-the
https://statistics.laerd.com/spss-tutorials/binomial-logistic-regression-using-spss-statistics.php

Hosmer-lemeshow 
https://cran.r-project.org/web/packages/generalhoslem/index.html

le Cessie-van Houwelingen-Copas-Hosmer
https://search.r-project.org/CRAN/refmans/DescTools/html/HosmerLemeshowTest.html
https://www.rdocumentation.org/packages/rms/versions/6.8-0/topics/residuals.lrm

**Refs**
Hosmer, D. W., Hosmer, T., Le Cessie, S., & Lemeshow, S. (1997). A Comparison of Goodness-of-Fit Tests for the Logistic Regression Model. Statistics in Medicine, 16(9), 965–980. [https://doi.org/10.1002/(SICI)1097-0258(19970515)16:9<965::AID-SIM509>3.0.CO;2-O](https://doi.org/10.1002/(SICI)1097-0258(19970515)16:9%3C965::AID-SIM509%3E3.0.CO;2-O)
Hosmer, D. W., & Lemesbow, S. (1980). Goodness of fit tests for the multiple logistic regression model. Communications in Statistics - Theory and Methods, 9(10), 1043–1069. https://doi.org/10.1080/03610928008827941
le Cessie, S., & van Houwelingen, J. C. (1991). A Goodness-of-Fit Test for Binary Regression Models, Based on Smoothing Methods. Biometrics, 47(4), 1267–1282. https://doi.org/10.2307/2532385
LEMESHOW, S., & HOSMER, D. W., JR. (1982). A REVIEW OF GOODNESS OF FIT STATISTICS FOR USE IN THE DEVELOPMENT OF LOGISTIC REGRESSION MODELS1. American Journal of Epidemiology, 115(1), 92–106. https://doi.org/10.1093/oxfordjournals.aje.a113284

**Comment by PerPalmgren** [comment]
Hi, Is there any advancement of the project to incorporate a linearity test such as Box-Tidwell for the logistic regression module?

**Comment by EJWagenmakers** [comment]
1. There has also been a suggestion for Shapiro Wilks in ANOVA. ([#3813](https://github.com/jasp-stats/jasp-issues/issues/3813#issuecomment-3434085712)) 
2. I like the idea of adding the Shapiro-Wilks test for normality in the Assumptions. However, what exactly does ANOVA assume? Does it assume normality of the residuals or normality in each cell of the design? @JohnnyDoorn? 
3. Even if ANOVA does not strictly require normality in each cell, it might be informative to show the Shapiro-Wilks results. But: should there be a correction for multiplicity? (each cell yields a test). Not sure how complicated this gets. How do other programs handle this? 
4. Did we make progress on any of these issues listed above? Don't we have Kolmogorov-Smirnov within the Distribution module? As an aside, we should always use the Lilliefors correction to the KS test (if the parameters of the distribution need to be estimated and are not known, which is usually the case)

**Comment by JohnnyDoorn** [comment]
Haha fair enough! I think the link is the other way around: if residuals are normally distributed, the cells will also be, but not necessarily the other way around, which is why it's better to test the residuals rather than the DV.

**Comment by JohnnyDoorn** [comment]
Hi all,

Love this discussion! I think the Q-Q plot gives a better assessment of the assumption violation on a continuum, rather than in the black-and-white kind of way that a NHST does, which I think is a better way of reasoning about statistical assumptions. However, I agree that we can add the S-W test to the assumption checks menu for completeness' sake, to make software transitions easier and facilitate teaching. In the short-term, as a workaround you can do this test by exporting the residuals from the ANOVA analysis, and then running the S-W test in Descriptives (which might even improve teaching the assumption).

Cheers,
Johnny

**Comment by EJWagenmakers** [comment]
I agree with Johnny: I prefer the QQ, but for completeness we can add the S-Ws. We aim to please :-)

**Comment by EJWagenmakers** [comment]
I just received another email about these residual tests. I am adding some more people to this issue to see whether there are any takers.

**Comment by tomtomme** [comment]
Breusch-Pagan-Test is now available as a community module and CIs for QQ-Plots are everywhere now too :)

Only missing:
- Box Tidwell
- QQ-plot for MANOVA
